### ISSUE TITLE
Exception while parsing remote OSGi Repository

### ISSUE THREAD

**Issue by amitjoy** [bndtools/bnd#5452]
The following exception occurs while parsing the following OSGi Repository resulting to all the other repositories not being loaded at all.

```
java.lang.reflect.InvocationTargetException
	at org.osgi.util.promise.DeferredPromiseImpl.getValue(DeferredPromiseImpl.java:128)
	at aQute.bnd.repository.osgi.OSGiIndex.getBridge(OSGiIndex.java:207)
	at aQute.bnd.repository.osgi.OSGiRepository.versions(OSGiRepository.java:194)
	at aQute.bnd.build.Project.getBundle(Project.java:1299)
	at aQute.bnd.build.Project.getBundles(Project.java:682)
	at aQute.bnd.build.Project.parseBuildpath(Project.java:578)
	at aQute.bnd.build.Project.prepare(Project.java:420)
	at aQute.bnd.build.Project.getDependson(Project.java:887)
	at aQute.bnd.build.Workspace.getBuildOrder(Workspace.java:747)
	at bndtools.central.Central.createWorkspace(Central.java:330)
	at aQute.bnd.memoize.MemoizingSupplier.lambda$new$1(MemoizingSupplier.java:31)
	at aQute.bnd.memoize.MemoizingSupplier.get(MemoizingSupplier.java:51)
	at bndtools.central.Central.getWorkspace(Central.java:230)
	at bndtools.central.Central.refreshPlugins(Central.java:548)
	at bndtools.views.repository.RepositoriesView$6.runInWorkspace(RepositoriesView.java:456)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:43)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
Caused by: org.osgi.util.promise.FailedPromisesException: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,16]
Message: found: DTD, expected START_ELEMENT or END_ELEMENT
	at org.osgi.util.promise.PromiseFactory$All.run(PromiseFactory.java:362)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,16]
Message: found: DTD, expected START_ELEMENT or END_ELEMENT
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.nextTag(XMLStreamReaderImpl.java:1361)
	at aQute.bnd.osgi.repository.XMLResourceParser.next(XMLResourceParser.java:193)
	at aQute.bnd.osgi.repository.XMLResourceParser.parse(XMLResourceParser.java:151)
	at aQute.bnd.repository.osgi.OSGiIndex.lambda$download$2(OSGiIndex.java:127)
	at org.osgi.util.promise.DeferredPromiseImpl$Map.accept(DeferredPromiseImpl.java:500)
	at org.osgi.util.promise.DeferredPromiseImpl.result(DeferredPromiseImpl.java:151)
	at org.osgi.util.promise.DeferredPromiseImpl$Map.run(DeferredPromiseImpl.java:492)
	... 3 more
Root exception:
org.osgi.util.promise.FailedPromisesException: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,16]
Message: found: DTD, expected START_ELEMENT or END_ELEMENT
	at org.osgi.util.promise.PromiseFactory$All.run(PromiseFactory.java:362)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: javax.xml.stream.XMLStreamException: ParseError at [row,col]:[1,16]
Message: found: DTD, expected START_ELEMENT or END_ELEMENT
	at java.xml/com.sun.org.apache.xerces.internal.impl.XMLStreamReaderImpl.nextTag(XMLStreamReaderImpl.java:1361)
	at aQute.bnd.osgi.repository.XMLResourceParser.next(XMLResourceParser.java:193)
	at aQute.bnd.osgi.repository.XMLResourceParser.parse(XMLResourceParser.java:151)
	at aQute.bnd.repository.osgi.OSGiIndex.lambda$download$2(OSGiIndex.java:127)
	at org.osgi.util.promise.DeferredPromiseImpl$Map.accept(DeferredPromiseImpl.java:500)
	at org.osgi.util.promise.DeferredPromiseImpl.result(DeferredPromiseImpl.java:151)
	at org.osgi.util.promise.DeferredPromiseImpl$Map.run(DeferredPromiseImpl.java:492)
	... 3 more
```

```
efxclipse.version                     =         3.8.0
efxclipse.url                         =         http://download.eclipse.org/efxclipse/runtime-released/${efxclipse.version}/site/repository.xml.gz
-plugin.5.efxclipse                   =         \
		    aQute.bnd.repository.osgi.OSGiRepository;\
		          locations           =         ${efxclipse.url};\
		          name                =         e(fx)clipse
```

Affected Versions: `6.4.0.REL` and `7.0.0.SNAPSHOT`

I checked the aforementioned OSGi repository if it is malformed but it seems to be fine and unchanged but still bndtools cannot parse it all of a sudden whereas it used to parse the repo earlier.

**Comment by bjhargrave** [comment]
I would suggest this is a download failure from download.eclipse.org. You will notice the parse failure occurs immediately on the first line which means the download result is probably some error text. 

You are using an `http` URL (`http://download.eclipse.org`) which is bad. Does this still fail for `https`?

Since you have a local copy of the index, does changing `efxclipse.url` to a `file` URL pointing to the local file still fail?

**Comment by amitjoy** [comment]
@bjhargrave You are right. Somehow, it could parse `http` earlier and now it cannot. It started working perfectly after changing the URL to use `https` instead.

**Comment by bjhargrave** [comment]
> Somehow, it could parse `http` earlier and now it cannot.

I think it is more the case that most web sites are shutting down `http` and only serving on `https`. Your use of `http` probably returned a message which said to use `https` instead. So there never was an issue on the Bnd side. It is about the `http` to `https` migration on the server side. You should review your use of `http` URLs everywhere with the view to replacing them with `https` URLs.

**Comment by bjhargrave** [comment]
> Your use of `http` probably returned a message which said to use `https` instead.

```
➜  ~ curl http://download.eclipse.org/efxclipse/runtime-released/3.8.0/site/repository.xml.gz
<html>
<head><title>301 Moved Permanently</title></head>
<body>
<center><h1>301 Moved Permanently</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Comment by amitjoy** [comment]
Thanks BJ for the pointers. I had updated the workspace configurations to use `https` instead of `http` 👍
